Common/src/interface_interpolation/CInterpolator.cpp

```diff
@@ -171,110 +171,117 @@ void CInterpolator::ReconstructBoundary(unsigned long val_zone, int val_marker){
-
-        if ( geom->nodes->GetVertex(dPoint, val_marker) != -1 )
-          nNodes++;
-      }
-
-      Buffer_Send_StartLinkedNodes[nLocalVertex] = nLocalLinkedNodes;
-      Buffer_Send_nLinkedNodes[nLocalVertex]     = nNodes;
-
-      nLocalLinkedNodes += nNodes;
-
-      Aux_Send_Map[nLocalVertex] = new unsigned long[ nNodes ];
-      nNodes = 0;
-
-      for (jEdge = 0; jEdge < nEdges; jEdge++){
-        EdgeIndex = geom->nodes->GetEdge(iPoint, jEdge);
-
-        if( iPoint == geom->edges->GetNode(EdgeIndex,0) )
-          dPoint = geom->edges->GetNode(EdgeIndex,1);
-        else
-          dPoint = geom->edges->GetNode(EdgeIndex,0);
+        Buffer_Send_Coord[iLocalVertex*nDim+iDim] = geom->nodes->GetCoord(iPoint, iDim);
+      neighbors.insert(pair<unsigned long, forward_list<unsigned long>*>(iPoint, new forward_list<unsigned long>));
+    }
+  }
 
-        if ( geom->nodes->GetVertex(dPoint, val_marker) != -1 ){
-          Aux_Send_Map[nLocalVertex][nNodes] = geom->nodes->GetGlobalIndex(dPoint);
-          nNodes++;
+  /*--- Define the neighbors map. ---*/
+  for(unsigned long iElem=0; iElem < nElems; iElem++){
+    CPrimalGrid* elem = geom->bound[val_marker][iElem];
+    for(unsigned short iNode=0; iNode<elem->GetnNodes(); iNode++){
+      iPoint = elem->GetNode(iNode);
+      if (geom->nodes->GetDomain(iPoint)) {
+        forward_list<unsigned long>* neighb = neighbors.at(iPoint);
+        for(unsigned short iNeighbor=0; iNeighbor<elem->GetnNeighbor_Nodes(iNode); iNeighbor++){
+          unsigned long jPoint = elem->GetNode( elem->GetNeighbor_Nodes(iNode,iNeighbor) );
+          unsigned long jPoint_global = geom->nodes->GetGlobalIndex(jPoint);
+          if( std::find(std::begin(*neighb), std::end(*neighb), jPoint_global) == std::end(*neighb) ){
+            neighb->emplace_front( jPoint_global );
+          }
         }
       }
-      nLocalVertex++;
     }
   }
 
-  unsigned long *Buffer_Send_LinkedNodes = new unsigned long [ nLocalLinkedNodes ];
-
+  // numbers of surface-neighbors of all domain vertices on the marker
+  unsigned long *Buffer_Send_nLinkedNodes       = new unsigned long [ nLocalVertex ];
+  // cumsum of Buffer_Send_nLinkedNodes
+  unsigned long *Buffer_Send_StartLinkedNodes   = new unsigned long [ nLocalVertex ];
   nLocalLinkedNodes = 0;
-
-  for (iVertex = 0; iVertex < nLocalVertex; iVertex++){
-    for (jEdge = 0; jEdge < Buffer_Send_nLinkedNodes[iVertex]; jEdge++){
-      Buffer_Send_LinkedNodes[nLocalLinkedNodes] = Aux_Send_Map[iVertex][jEdge];
-      nLocalLinkedNodes++;
+  for (iVertex = 0; iVertex < nVertex; iVertex++) {
+    iPoint = geom->vertex[val_marker][iVertex]->GetNode();
+    if (geom->nodes->GetDomain(iPoint)) {
+      unsigned long iLocalVertex = iVertex_to_iLocalVertex[iVertex];
+      Buffer_Send_nLinkedNodes[iLocalVertex] = std::count_if(
+        std::begin(*neighbors[iPoint]), std::end(*neighbors[iPoint]),
+        [](unsigned long i){return true;} );
```

Comment:
And what are we doing here? Counting the elements in a list? Doesn't look very expressive :confused:

Reply:
You're right, this is fixed in 480d9a3 also